packages/shared/ReactErrorUtils.js

```diff
@@ -185,6 +190,9 @@ if (__DEV__) {
         // nested call would trigger the fake event handlers of any call higher
         // in the stack.
         fakeNode.removeEventListener(evtType, callCallback, false);
+        if (windowEvent && window.event !== windowEvent) {
+          window.event = windowEvent;
```

Comment:
Have you verified that the browsers that support `window.event` define it as a writable property?

Reply:
I've verified that the fixture works in Chrome, Safari and IE 11. Firefox does not support this. Is there a complete list of browsers I should try and support, or any support for running the tests/fixtures in a large set of real browser environments?

Reply:
It might be good to test in IE9 and IE10 as well if they support it.

Does this get properly reset after the event dispatch is complete? We need to be sure that `window.event` doesn't start persisting, since it should only be defined while an event is in progress.

Reply:
Good call on checking. It does not work in IE 9 or 10 or 11 (I'm not sure how I persuaded myself that it did before), as it fails with "Assignment to read-only properties is not allowed in strict mode".

The event does get reset correctly in Chrome and Safari, and I've added a check for that to the fixture.

There are two other approaches I considered, but neither seem to work:

* Instead of `window.event = `, defining a property on window that returns the correct event. This works to return the correct event object (as far as I can tell), but the event object is unusable in IE 9 or 10 (it does work in IE 11) because any attempt to access the event's properties throws "Member not found".
* Instead of creating a new fake event to dispatch, just re-dispatching window.event. (this fails with an error that "event is already being dispatched")

For now I've updated this pull request to not preserve window.event in IE at all. If it would be preferred, I can make it work in IE 11 using `window.defineProperty` — it still will not work in IE9 or 10, but it can be made to not crash.